### DISCUSSION TITLE
Visiting localhost:3036 yields an empty page

### DISCUSSION THREAD

**Discussion by rsieb** [ElMassimo/vite_ruby#320] [Q&A]
- [x] I have tried upgrading by running `bundle update vite_ruby`.
- [x] I have read the __[troubleshooting section]__ before opening an issue.

### Description 📖

Several apps with vite_rails produce a blank html page and no console message after starting `bin/vite dev` 

STDOUT just says:
````
  VITE v4.0.1  ready in 369 ms

  ➜  Local:   http://127.0.0.1:3036/vite-dev/
  ➜  press h to show help

````

Connecting to http://127.0.0.1:3036/vite-dev/ produces the blank html, and on manual reload only, the console says 404 'not found'. See screenshots below. 

![CleanShot 2022-12-15 at 19 28 51](https://user-images.githubusercontent.com/3195281/208015773-5f7887ee-6e6d-41d0-9ba7-d3529bebfb4a.png)
![CleanShot 2022-12-15 at 19 29 17](https://user-images.githubusercontent.com/3195281/208015786-2dc39bf9-0da7-4aa5-94c7-4fb9aa00dbd8.png)

Connecting to http://127.0.0.1:3000 or http://127.0.0.1:5000/ produces an "unable to connect" message, see screenshot below

![CleanShot 2022-12-15 at 19 30 51](https://user-images.githubusercontent.com/3195281/208015810-e05bb9c3-ed5a-4c5e-b421-dec9e06ec70e.png)

The same bug appears across several rails apps, including the example app and one other where vite_rails previously worked fine. I have confirmed it also occurs on a brand new fresh rails app with all the defaults and 'vite_rails' as the only extra gem.

I wonder if this has to do with a recent major upgrade of vite? I'm not sure I remember vite 4.0.1 coming up when it was working previously but maybe I just missed it. 

### Reproduction 🐞

_Please provide a link to a repo that can reproduce the problem you ran into._

https://github.com/rsieb/viterailsclean.git

- run bundle/setup
- run bin/vite dev
- see errors above

<summary>Vite Ruby Info</summary>

<details>
_Run `bin/rake vite:info` and provide the output:_

```
> bin/rake vite:info
bin/vite present?: true
vite_ruby: 3.2.13
vite_rails: 3.0.13
rails: 7.0.4
node: v16.17.0
npm: 8.15.0
yarn: 1.22.19
pnpm: 7.16.1
ruby: ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [arm64-darwin21]

installed packages:
viterailsclean@ /Users/rsieb/gits/viterailsclean
├─┬ vite-plugin-ruby@3.1.3
│ └── vite@4.0.1 deduped
└── vite@4.0.1
```
</details>

### Logs 📜

_If not providing a reproduction:_

<details>
<summary>Output</summary>

_Run `DEBUG=vite-plugin-ruby:* bin/vite dev` or `DEBUG=vite-plugin-ruby:* bin/vite build` and provide the output:_

```
VITE v4.0.1  ready in 235 ms

  ➜  Local:   http://localhost:3036/vite-dev/
  ➜  press h to show help
```
</details>

### Screenshots 📷

_Provide console or browser screenshots of the problem_.

See above

**Comment by ElMassimo** [comment]
Hi Roland!

As mentioned in the [guide](https://vite-ruby.netlify.app/guide/#running-your-first-example-%F0%9F%8F%83%E2%80%8D%E2%99%82%EF%B8%8F), there are two steps to get started:

- Run `bin/vite dev` to start the Vite development server
- Restart your Rails web server before visiting any page

It seems that you are starting Vite, but not Rails, which is why you don't see anything in `localhost:3000`.

As to `localhost:3036` which is Vite, you should not visit it directly, as it has no html to serve in most Ruby setups.

---

Thanks for [providing a repo](https://github.com/rsieb/viterailsclean.git).

When I run `bin/rails s` and visit `localhost:3000`:

<img width="1414" alt="Screen Shot 2022-12-16 at 08 45 38" src="https://user-images.githubusercontent.com/1158253/208091406-eb4002c4-82ff-4e13-b5b3-35af6bcb675f.png">

If I add my first page to Rails, and visit it when running `bin/vite dev`:

<img width="918" alt="Screen Shot 2022-12-16 at 08 51 22" src="https://user-images.githubusercontent.com/1158253/208092311-4db6c50c-7656-4310-81e1-4f77677bc684.png">
